WeatherMonitoringSystem/Domain/Core/WeatherStation.cs

```diff
@@ -0,0 +1,43 @@
+﻿using WeatherMonitoringSystem.Bots;
+
+namespace WeatherMonitoringSystem.Core
+{
+    public class WeatherStation
+    {
+        private List<IWeatherBot> _bots = new();
+        private WeatherData _currentWeather;
+
+        public void AddBot(IWeatherBot bot)
+        {
+            _bots.Add(bot);
+        }
+
+        public void RemoveBot(IWeatherBot bot)
+        {
+            _bots.Remove(bot);
+        }
+
+        public List<IWeatherBot> SetWeatherData(WeatherData data)
+        {
+            _currentWeather = data;
+            return GetActivatedBots();
+        }
+
+        private List<IWeatherBot> GetActivatedBots()
```

Comment:
renaming the method to Getxxx is still confusing, it hides with the method is actually doing, the method is doing to thing: triggering the bot + returning active bots. 
Just separate to two methods as suggested before, or return all bots from this method. It's important for the method name to be clear enough, it's fine to return all bots , but if returns just partial list , the caller can't tell that this method is doing that.